Feature/56 add azure docs (#59)
tamu-edu/it-cloud-docs: src/cloud/azure/network/design.md

Cloud Services operates a shared instance of Azure Front Door that can be used by all customers in the TAMU-managed network. Customers can request to have their web applications or services added as origins to the shared AFD instance, and Cloud Services will configure the necessary routing and WAF policies to ensure secure access. Custom domains are supported and certificates will be managed by AFD and Cloud Services with automatic renewals.

Currently, this is a manual request and configuration process, but in the future we plan to automate this through self-service tools and integration with Azure services.

### Azure Firewall

Azure Firewall is a stateful firewall service that provides network and application-level protection for resources in Azure. It is used in the TAMU-managed network design to implement required rules to implement security controls, such as geoblocking and other targeted restrictions, and provide connectivity for non-web workloads that require public access.

The firewall is also used to route traffic between Azure virtual networks and campus networks.

> Customers can request to have firewall rules added to the shared Azure Firewall instance to allow for secure access to their resources from the internet or other networks. Currently, this is a manual request and configuration process, but in the future we plan to automate this through self-service tools and integration with Azure services.

Author: I feel this is in contrast to another comment in Access Methods doc that there will be no exceptions to Firewall rules. One or both statements should be clarified.

Reviewer: the exceptions are just for admin or unencrypted ports. It doesn't explicitly say we'll open up RDP here, but it can be clearer.

## Provisioning and Configuration

Customer VNets ("spoke networks") are provisioned by Cloud Services with one or more IP address ranges (CIDR) allocated to it from a pool of address space. The default allocation will be 32 IP addresses (27 usable), but can be grown as needed.

Our default Azure region is `South Central US`, as it is closest to the TAMU campus and a majority of our user base and will have the lowest latency, and virtual networks will be provisioned in this region unless otherwise requested. Follow the guidance in [creating subnets](creating_subnets.md) for dividing your address space into subnets and connecting resources to those subnets. If you need to, work with Cloud Services to ensure your VNet address space and subnet design align with your solution's need and recommended best practices.

By default, no subnets will be created in spoke VNets since subnet design can vary greatly based on the types of resources being deployed. The customer is responsible for creating and configuring their own subnets within their VNet, but Cloud Services can provide guidance and best practices for doing so.

> All subnets will be private with no direct access to or from the public internet. A user-defined route (UDR) will be created to route outbound internet traffic through the Azure Firewall and NAT Gateway services in the hub VNet and will need to be associated with each subnet. An Azure policy will be put in place to prevent the creation of public subnets, and to attach the default UDR to all subnets missing one.

Author: ```suggestion
All subnets shall be private with no direct access to or from the public internet. A user-defined route (UDR) shall be created to route outbound internet traffic through the Azure Firewall and NAT Gateway services in the hub VNet and will need to be associated with each subnet. An Azure policy will be put in place to prevent the creation of public subnets, and to ensure that all subnets have the default UDR attached.
```
